infrastructure/io/src/main/java/tech/pegasys/teku/infrastructure/io/SyncDataAccessor.java

```diff
@@ -66,4 +116,23 @@ public void syncedWrite(final Path path, final Bytes data) throws IOException {
         StandardOpenOption.CREATE,
         StandardOpenOption.TRUNCATE_EXISTING);
   }
+
+  private static void atomicSyncedWrite(final Path path, final Bytes data)
+      throws AtomicMoveNotSupportedException {
+    final Path absolutePath = path.toAbsolutePath();
+    final Path tmpFile = Paths.get(path.toString() + ".tmp");
+    try {
+      nonAtomicSyncedWrite(tmpFile, data);
+      Files.move(
+          tmpFile,
+          absolutePath,
+          StandardCopyOption.ATOMIC_MOVE,
+          StandardCopyOption.REPLACE_EXISTING);
+    } catch (AtomicMoveNotSupportedException e) {
+      throw e;
+    } catch (IOException e) {
+      LOG.error(String.format("Failed to write in %s", path), e);
+      throw new InvalidConfigurationException(String.format("Cannot write to folder %s", path), e);
+    }
```

Comment:
I think we should just be throwing `IOException` from here so that the behaviour matches `nonAtomicSynedWrite`.

We'd catch `IOException` in the `create` method and convert it to `InvalidConfigurationException` there so it only applies during the startup check.